src/internal/_toString.js

```diff
@@ -0,0 +1,36 @@
+var _indexOf = require('./_indexOf');
+var _map = require('./_map');
+var _quote = require('./_quote');
+var _toISOString = require('./_toISOString');
+var keys = require('../keys');
+
+
+module.exports = function _toString(x, seen) {
+    var recur = function recur(y) {
+        var xs = seen.concat([x]);
+        return _indexOf(xs, y) >= 0 ? '<Circular>' : _toString(y, xs);
+    };
+
+    switch (Object.prototype.toString.call(x)) {
+        case '[object Arguments]':
+            return '(function() { return arguments; }(' + _map(recur, x).join(', ') + '))';
+        case '[object Array]':
+            return '[' + _map(recur, x).join(', ') + ']';
+        case '[object Boolean]':
+            return typeof x === 'object' ? 'new Boolean(' + recur(x.valueOf()) + ')' : x.toString();
+        case '[object Date]':
+            return 'new Date(' + _quote(_toISOString(x)) + ')';
+        case '[object Null]':
+            return 'null';
+        case '[object Number]':
+            return typeof x === 'object' ? 'new Number(' + recur(x.valueOf()) + ')' : 1 / x === -Infinity ? '-0' : x.toString(10);
+        case '[object String]':
+            return typeof x === 'object' ? 'new String(' + recur(x.valueOf()) + ')' : _quote(x);
+        case '[object Undefined]':
+            return 'undefined';
+        default:
+            return x.toString !== Object.prototype.toString ?  // Boolean, Function, RegExp, user-defined types
```

Comment:
I haven't yet worked out how to *test* cross-frame compatibility. I served a document from `http://localhost:4000` containing an iframe served from `http://localhost:4001`, and used [`window.postMessage`][1] to send values to the iframe, but `instanceof` checks worked correctly.


[1]: https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage